### PR TITLE
feat(otrm): Support validating filter processors in the source templates

### PR DESCRIPTION
**Pull request by rnishtala-sumo**
Support validating filter and transform processors in the source templates

### CODE REVIEW

File: pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+          - windowseventlog/Microsoft-Windows-PowerShell/Operational/09c98ea4-6686-4ab9-b3cd-f04070906441
+          - windowseventlog/Amazon EC2Launch/09c98ea4-6686-4ab9-b3cd-f04070906441
+        processors:
+          - memory_limiter
+          - resourcedetection/system
+          - batch
+          - resource/09c98ea4-6686-4ab9-b3cd-f04070906441
+          - filter/logs/09c98ea4-6686-4ab9-b3cd-f04070906441
+        exporters:
+          - sumologic
```
Comment (sumo-drosiek) [inline pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml:103]: Do we need all of this configuration just to test filterprocessor?

Comment (rnishtala-sumo) [reply pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml:103]: this is an example source template from the backend team, that I thought would be useful to run a test against.

Comment (sumo-drosiek) [reply pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml:103]: I feel like it is more than needed. Personally I would like to limit test scope to what is necessary, eventually we can add another test which will be more complex, but now most of the configuration seems to be unrelated to the name of the test/file

Comment (sumo-drosiek) [reply pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml:103]: If the test fail, it won't be obvious why

Comment (rnishtala-sumo) [reply pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml:103]: ok I'll trim some portions of the config.

Comment (rnishtala-sumo) [reply pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml]: how does this look - https://github.com/SumoLogic/sumologic-otel-collector/pull/1547/commits/7fcad3e5406cebd3fb5e8bf231ab66953b3f8851

it has 1 receiver, 2 filter processors, 1 transform processor, 1 exporter.

File: pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+  receivers:
+    hostmetrics/09c98ea4-6686-4ab9-b3cd-f04070906441:
+      collection_interval: 5m
+      scrapers:
+        network:
+  processors:
+    batch:
+      send_batch_size: 1_024
+      timeout: 1s
```
Comment (sumo-drosiek) [inline pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml:9]: 🙈

File: pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml (hunk 1 above)
Comment (sumo-drosiek) [reply pkg/extension/opampextension/testdata/opamp.d/opamp-filter-processor.yaml]: Yup, thats better, LGTM :) (I marked batch processor, you can clean it up if unnecessary)

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**